sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -141,19 +141,46 @@ static void _on_sock_evt(sock_udp_t *sock, sock_async_flags_t type, void *arg)
     sock_udp_ep_t remote;
 
     if (type & SOCK_ASYNC_MSG_RECV) {
-        ssize_t res = sock_udp_recv(sock, _listen_buf, sizeof(_listen_buf),
-                                    0, &remote);
-        if (res <= 0) {
-            DEBUG("gcoap: udp recv failure: %d\n", (int)res);
-            return;
+        void *stackbuf;
+        void *buf_ctx = NULL;
+        bool truncated = false;
+        size_t cursor = 0;
+
+        /* The zero-copy _buf API is not used to its full potential here -- we
+         * still copy out data in what is a manual version of sock_udp_recv,
+         * but this gives the direly needed overflow information.
+         *
+         * A version that actually doesn't copy would vastly change the way
+         * gcoap passes the buffer to be read from and written into to the
+         * handler. Also, given that neither nanocoap nor the handler expects
+         * to gather scattered data, it'd need to rely on the data coming in a
+         * single slice (but that may be a realistic assumption).
+         */
+        while (true) {
+            ssize_t res = sock_udp_recv_buf(sock, &stackbuf, &buf_ctx,
+                                        0, &remote);
```

Comment:
```suggestion
            ssize_t res = sock_udp_recv_buf(sock, &stackbuf, &buf_ctx,
                                            0, &remote);
```

If this also fits the 100 char limit, no need to break the line